store/tikv/region_cache.go

```diff
@@ -581,26 +579,6 @@ func (r *Region) GetContext() *kvrpcpb.Context {
 	}
 }
 
-// OnRequestFail records unreachable peer and tries to select another valid peer.
-// It returns false if all peers are unreachable.
-func (r *Region) OnRequestFail(storeID uint64) bool {
```

Comment:
Why remove this? Is this logic useless or moved to another place?

Reply:
It's renamed to `DropStoreOnSendRequestFail`

Reply:
It's useless.

Reply:
There are some considerations for using unreachable store list.

Consider a store is down, another peer of the region becomes the leader, but somehow the new leader is not able to send heartbeat to PD in time.

With the unreachable store list, tidb can try the other peers automatically. Otherwise, it will continue to reconnect the down tikv until it times out